ceph_iscsi_config/gateway_setting.py

```diff
@@ -0,0 +1,167 @@
+                             format(self.max_val, self.name))
+        return val
+
+
+CLIENT_SETTINGS = {
+    "dataout_timeout": IntSetting("dataout_timeout", 2, 60, 20),
+    "nopin_response_timeout": IntSetting("nopin_response_timeout", 3, 60, 5),
+    "nopin_timeout": IntSetting("nopin_timeout", 3, 60, 5),
+    "cmdsn_depth": IntSetting("cmdsn_depth", 1, 512, 128)}
+
+TGT_SETTINGS = {
+    # client settings you can also set at the ceph-iscsi target level
+    "dataout_timeout": IntSetting("dataout_timeout", 2, 60, 20),
+    "nopin_response_timeout": IntSetting("nopin_response_timeout", 3, 60, 5),
+    "nopin_timeout": IntSetting("nopin_timeout", 3, 60, 5),
+    "cmdsn_depth": IntSetting("cmdsn_depth", 1, 512, 128),
+    # lio tpg settings
+    "immediate_data": LIOBoolSetting("immediate_data", True),
+    "initial_r2t": LIOBoolSetting("initial_r2t", True),
+    "max_outstanding_r2t": IntSetting("max_outstanding_r2t", 1, 65535, 1),
+    "first_burst_length": IntSetting("first_burst_length", 512, 16777215, 262144),
+    "max_burst_length": IntSetting("max_burst_length", 512, 16777215, 524288),
+    "max_recv_data_segment_length": IntSetting("max_recv_data_segment_length",
+                                               512, 16777215, 262144),
+    "max_xmit_data_segment_length": IntSetting("max_xmit_data_segment_length",
+                                               512, 16777215, 262144)}
+
+SYS_SETTINGS = {
+    "cluster_name": StrSetting("cluster_name", "ceph"),
+    "pool": StrSetting("pool", "rbd"),
+    "cluster_client_name": StrSetting("cluster_client_name", "client.admin"),
+    "time_out": IntSetting("time_out", 1, 600, 30),
+    "api_host": StrSetting("api_host", "::"),
+    "api_port": IntSetting("api_port", 1, 65535, 5000),
+    "api_secure": BoolSetting("api_secure", True),
+    "api_ssl_verify": BoolSetting("api_ssl_verify", False),
+    "loop_delay": IntSetting("loop_delay", 1, 60, 2),
+    "trusted_ip_list": ListSetting("trusted_ip_list", []),  # comma separate list of IPs
+    "api_user": StrSetting("api_user", "admin"),
+    "api_password": StrSetting("api_password", "admin"),
+    "ceph_user": StrSetting("ceph_user", "admin"),
+    "debug": BoolSetting("debug", False),
+    "minimum_gateways": IntSetting("minimum_gateways", 2, 9999, 2),
+    "ceph_config_dir": StrSetting("ceph_config_dir", '/etc/ceph'),
+    "priv_key": StrSetting("priv_key", 'iscsi-gateway.key'),
+    "pub_key": StrSetting("pub_key", 'iscsi-gateway-pub.key'),
+    "prometheus_exporter": BoolSetting("prometheus_exporter", True),
+    "prometheus_port": IntSetting("prometheus_port", 1, 65535, 9287),
+    "prometheus_host": StrSetting("prometheus_host", "::"),
+    "logger_level": IntSetting("logger_level", logging.DEBUG, logging.CRITICAL,
+                               logging.DEBUG),
+    # TODO: This is under sys for compat. It is not settable per device/backend
+    # type yet.
+    "alua_failover_type": StrSetting("alua_failover_type", "implicit")}
```

Comment:
What about having an `EnumSetting` that knows about the list of possible values (and can validate the value based on that)?

Something like:
```
EnumSetting("alua_failover_type", "implicit", ["implicit", "other_valid_value"])
```

Reply:
Nice. Done.